src/Client.php

```diff
@@ -141,21 +141,59 @@ public function create(AutotaskObjects\Entity $obj)
         $params = new AutotaskObjects\CreateParam($obj);
         return $this->_call('create', array($params));
     }
+    
+    public function bulkCreate(array $objs)
+    {
+        if (count($objs) > 200) {
+            throw new \Exception('You can only execute a bulk create on a max of 200 objects per request');
+        }
+        $createObjs = null;
+        foreach ($objs as $obj) {
+            if (!$createObjs){
+               $createObjs = new AutotaskObjects\CreateParam($obj);
+            } 
+            else {
+               $createObjs->Entities[] = $obj;
+            }
+        }
+        return $this->_call('create', array($createObjs));
+    }    
 
     public function update(AutotaskObjects\Entity $obj)
     {
         $params = new AutotaskObjects\UpdateParam($obj);
         return $this->_call('update', array($params));
     }
-
-    public function bulkUpdate(array $objs)
-    {
-        $updateObjs = array();
+    
+    public function bulkUpdate(array $objs) {
+	if (count($objs) > 200) {
+            throw new \Exception('You can only execute a bulk update on a max of 200 objects per request');
+        }
+        $updateObjs = null;
+        $calls = array();
         foreach ($objs as $obj) {
-            $updateObjs[] = new AutotaskObjects\UpdateParam($obj);
+            if (!$updateObjs) {
+                $updateObjs = new AutotaskObjects\UpdateParam($obj);
+            }
+            else {
+                $updateObjs->Entities[] = $obj;
+            }
         }
-
-        return $this->_call('update', $updateObjs);
+        return $this->_call('update', array($updateObjs));
+    }
+    
+    public function bulkUpdateUnlimited(array $objs) {
```

Comment:
I still don't really like pushing this logic down to the library.

The issue now becomes memory use. What if I push in 1000 updates without considering it, and the result sets (which are pretty huge for bulk actions) cause a memory allocation error?

This is logic that should exist outside of the library where we dont have to store all of the responses, but can loop over the `bulkUpdate` call, and operate on each response as it happens, therefore saving quite a bit of memory.